plugins/catalog/src/components/AboutCard/fields.tsx

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getEntitySourceLocation,
+  RELATION_OWNED_BY,
+  RELATION_PART_OF,
+} from '@backstage/catalog-model';
+import {
+  EntityRefLinks,
+  getEntityRelations,
+  useEntity,
+} from '@backstage/plugin-catalog-react';
+import { JsonArray } from '@backstage/types';
+import { Chip, makeStyles, Typography } from '@material-ui/core';
+import React from 'react';
+import { AboutField } from './AboutField';
+import { LinksGridList } from '../EntityLinksCard/LinksGridList';
+
+const useStyles = makeStyles({
+  description: {
+    wordBreak: 'break-word',
+  },
+});
+
+function getLocationTargetHref(
+  target: string,
+  type: string,
+  entitySourceLocation: {
+    type: string;
+    target: string;
+  },
+): string {
+  if (type === 'url' || target.includes('://')) {
+    return target;
+  }
+
+  const srcLocationUrl =
+    entitySourceLocation.type === 'file'
+      ? `file://${entitySourceLocation.target}`
+      : entitySourceLocation.target;
+
+  if (type === 'file' || entitySourceLocation.type === 'file') {
+    return new URL(target, srcLocationUrl).href;
+  }
+
+  return srcLocationUrl;
+}
+
+/**
+ * The built-in fields to be used individually.
+ *
+ * @public
+ */
+export const AboutCardBuiltInFields = Object.freeze({
```

Comment:
I think that it's an issue because it's an extension when you import it, it's not the actual component. would maybe we can do some typescript magic to make it work.

Reply:
Yes, we had a discussion here https://github.com/backstage/backstage/pull/12651#discussion_r921946063. I didn't find a way to export it under `EntityAboutCard`, is there some magic that can help me achieve this? 🪄

Reply:
Ah alright! Cool, we had a discussion just now and indeed that's a tricky one since we haven't built in support yet for transferring over the static fields. Would you mind trying to do it manually and see how it works? I.e., right after `catalogPlugin.provide(createComponentExtension(...`, assign the fields to the output just like for the inner AboutCard. The important thing is to check whether typescript automatically is smart enough to realize that those fields now exist on the output thing when you import it elsewhere. So you'd want to exercise `EntityAboutCard.fields` in a test to see that it is picked up.

Essentially you'd have both of these

```ts
// in AboutCard.tsx
import { aboutCardFields, defaultAboutCardFields } ...
AboutCard.fields = aboutCardFields;
AboutCard.defaultFields = defaultAboutCardFields;
// in plugin.ts
import { aboutCardFields, defaultAboutCardFields } ...
EntityAboutCard.fields = aboutCardFields;
EntityAboutCard.defaultFields = defaultAboutCardFields;
```

If plain assignment doesn't work, try Object.assign (in further attempts to appease typescript).

Unfortunately this means that the fields will still not be lazily loaded, but maybe this is a good middle ground until we can invent a better way of making this happen automatically in createComponentExtension.

Reply:
Hi @freben, I've tried both and can't get it to work, unfortunately. `Object.assign` doesn't throw an error, but I couldn't access the fields in `EntityAboutCard`.

Reply:
Yep we'll have to abandon this train of thought indeed, since the lazy loading of provided components throw a wrench into things as well.